Automattic/jetpack
# Membership Block

## This is the first PR of the block that is intended to provide something that beta users can test. We intend to work in small iterations and we will gladly address feedback in later PRs. 

This introduces membership-button block to Jetpack.
This essentially will be a recurring donation button.

Purpose of the code in this PR is to:

- introduce endpoints that communicate with WPCOM
- whitelist certain options, CPTs and meta to store / sync data,
- introduce gutenberg block that uses these endpoints and provides UI to connect to Stripe, create and choose a product
- introduce a frontend of a block with the sole purpose of displaying a checkout window from wpcom in an iframe

~This needs D26086-code .~ - deployed


## Screenshots

![Zrzut ekranu 2019-03-27 o 18 17 26](https://user-images.githubusercontent.com/3775068/55098446-0df9e500-50be-11e9-99fb-5ffa6b6b55b9.png)


![Zrzut ekranu 2019-03-27 o 17 41 59](https://user-images.githubusercontent.com/3775068/55097068-64195900-50bb-11e9-8997-2d05bdc70dcc.png)

![Zrzut ekranu 2019-03-27 o 17 47 59](https://user-images.githubusercontent.com/3775068/55097075-6976a380-50bb-11e9-970a-853da7abf6b8.png)

## Internal references:

p89M8K-4a-p2
p1HpG7-6n6-p2

## Testing instructions - Jurassic Ninja

The problem with this is that you can only use real money and real stripe accounts. If you want sandbox payments, you need to use the ngrok flow

1. https://jurassic.ninja/create/?jetpack-beta&branch=add/memberships
2. Connect Jetpack, enable "BETA_BLOCKS" in /wp-admin/options-general.php?page=companion_settings
3. Go to editor
4. Search for "Membership button" block
5. You will get a placeholder that will say "connect to stripe", click that
6. If you correctly set up the store sandbox, you will get "you can skip this form" at the top, click that
7. You will get redirected to ugly looking page on success
8. Now go back to gutenberg, click "Re-check Connection", wait
9. Now you should have a button to add product, click it, fill the form
10. choose your new product
11. Publish the page
12 Go to frontend, click the button that says contribution
13. go with the flow

## Testing instructions - Jetpack Docker

1. Set up Jetpack with docker and ngrok.io, connnect jetpack
2. Remember to turn on beta blocks
```
define( 'GUTENBERG_DEVELOPMENT_MODE', true );
define( 'JETPACK_BETA_BLOCKS', true );
```
3. Checkout this PR
4. Build blocks by
```
yarn build-extensions
```
5. Continue from point 3. in the previous instruction
